I have added another attack mode to make it seem a bit more human like

foolchan2556 opened the pull request:
But I m not quite experienced with open source work flow, hope I won't mess everything up :)

foolchan2556 committed:
Add another attack mode attackashumanbeings

foolchan2556 committed:
add another attack mode Attack4SideAsHumanbeings

norecha commented:
Thanks man, I'm on vacation now, I will check it out when I'm back

code-borker commented:
So I played with it, merging the current 1.4.0 snapshot master branch here, and your PR.

To start, I *love* its drop patterns, very well done.  Only way I could see someone seeing it as a bot is when it drops troops in a location where a person wouldn't normally do so.

That said, it seems to break when you bring a filled CC.  It keeps that in the troop count.  It constantly is attempting to use the AQ and BK abilities after they've been used.  Maybe detect the usage, and don't try again.

It does give itself away when it's sitting there doing nothing after attacking, no more units to deploy, yet it's still attempting to drop barbs, and archers.

I've poked around it a bit, but java is one of my weaker languages, and it can be hard for me to follow another persons' thought processes through these types of programs, so I don't have any proposed fixes.

However, it would be awesome (and it seems like it's possible), to have the bot analyze the base, see where the collectors, and maybe even storages are, and orient drops to get at them.

foolchan2556 commented:
Hi @code-borker , thanks for your feedback! It could be my bad that I changed the dropping threshold a little bit, which might keep the bot clicking in vain. I have fixed that in my local version but not yet submitted another PR.

All we need to do is to change the line 45 of Attack4sideAsHumanbeings to while (troopCountsSum > 1), which will prevent it from dropping in crazy. If you are convenient, could you fix that, if not I will submit another PR. Thanks a lot!

I strongly agree with your points about the further development, and I believe it could be fantastic if only we can detect the red line of the bases when attacking, we can make it much more human like. Currently pokubot still behaves in a robot way so that we have to tradeoff some farming efficiency to avoid being too straightforward.  But once we can detect border line, we could farm as much as we want and that will be undeniable lol!

foolchan2556 commented:
wait a minute, @code-borker are you @norecha ? If you don't own the repo, don't bother to fix that bug, I can do it myself. Thanks :)

code-borker commented:
I am not norecha.  Thanks for the fix idea, I can get in and change my local copy too and recompile, thanks

kingburat commented:
Pls support droid4x   can i use this

akeyy commented:
@norecha where/how  to put the codes ?  newbie here :)

erik-grabljevec commented:
Hello everyone! I would love to join this project. Is anyone currently active here?
